client/comms/wsconn.go

```diff
@@ -0,0 +1,360 @@
+package comms
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/decred/dcrdex/server/comms/msgjson"
+	"github.com/gorilla/websocket"
+)
+
+const (
+	// bufferSize is buffer size for the websocket connection's read channel.
+	readBuffSize = 128
+)
+
+// WsCfg is the configuration struct for initializing a WsConn.
+type WsCfg struct {
+	// The websocket host.
+	Host string
+	// The websocket api path.
+	Path string
+	// The maximum time in seconds to wait for a ping from the server.
+	PingWait time.Duration
```

Comment:
One possibility is to not have this wait timeout at all on the client side, or maybe have the client ping too.  Although I suspect it would be best to either agree on this duration, or communicate it after connect.